src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -12,23 +12,70 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
 
 public class ProfileServlet extends HttpServlet {
-	//implement doGet so it knows where to go
+	/** Store class that gives access to Users. */
+	private UserStore userStore;
+	/**
+	 * Set up state for handling login-related requests. This method is only called when running in a
+	 * server, not when running in a test.
+	 */
+	@Override
+	public void init() throws ServletException {
+		super.init();
+		setUserStore(UserStore.getInstance());
+	}
+
+	/**
+	 * Sets the UserStore used by this servlet. This function provides a common setup method for use
+	 * by the test framework or the servlet's init() function.
+	 */
+	void setUserStore(UserStore userStore) {
+		this.userStore = userStore;
+	}
 	@Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
 		String requestUrl = request.getRequestURI();
 		String currentProfile = requestUrl.substring("/user/".length());
-		String username = (String) request.getSession().getAttribute("user");
-
-		if (currentProfile.equals(username)){
```

Comment:
Was this previously just used for debugging purposes?

Reply:
This was part of an unnecessarily complicated this I was doing. Instead of using just one profile.jsp file, I was using one for when a user was seeing their own profile page and one for when they weren't. I figured out how to use just one jsp file hence why I deleted these lines.

Reply:
Ahh okay cool